service/matching/backlog_manager.go

```diff
@@ -298,3 +299,35 @@ func (c *backlogManagerImpl) newIOContext() (context.Context, context.CancelFunc
 func (c *backlogManagerImpl) queueKey() *PhysicalTaskQueueKey {
 	return c.pqMgr.QueueKey()
 }
+
+// countTasksFromTaskQueue counts the total number of tasks present in a task queue; returns 0 for
+// cassandra databases since the in-memory counter is a good estimate for the number of tasks present while it returns
+// the actual number of tasks in the task queue for sql databases. This is required since SQL databases do not persist
+// taskQueueInfo when tasks are created.
+func (c *backlogManagerImpl) countTasksFromTaskQueue(ctx context.Context) (int64, error) {
+	tasksPresent, err := c.db.store.CountTasksExact(ctx, &persistence.CountTasksExactRequest{
+		NamespaceID: c.db.queue.NamespaceId().String(),
+		TaskQueue:   c.db.queue.PersistenceName(),
+		TaskType:    c.db.queue.TaskType(),
+	})
+	if _, ok := err.(*serviceerror.Unimplemented); ok {
+		tasksPresent = 0
+	} else if err != nil {
+		return 0, err
+	}
+	return int64(tasksPresent), nil
+}
+
+func (c *backlogManagerImpl) getApproximateBacklogCount(ctx context.Context) (int64, error) {
```

Comment:
I think above this method we should describe the situations that we may under or over count. (when CountTasksExact is not used). These situations comes to my mind:
- Overcount: backlog manager is killed without having the chance to persist latest ack level is
- Undercount: ownership transfers and the old owner writes more tasks to backlog between the GetTaskQueue and UpdateTaskQueue calls of the new owner lease takeover.
- Undercount: the scenario which moving ackLevel to db would have solved

Reply:
I think for point (1), you mean the situation in which Cassandra randomly TTL'es out tasks which in-turn leads to the counter being an overestimate right?